lib/core/src/msParam.cpp

```diff
@@ -1451,60 +1451,61 @@ int
 addKeyValToMspStr( msParam_t * keyStr, msParam_t * valStr,
                    msParam_t * msKeyValStr ) {
 
-    if ( ( keyStr == NULL && valStr == NULL ) || msKeyValStr == NULL ) {
+    if ((keyStr == nullptr && valStr == nullptr) || msKeyValStr == nullptr) {
         return SYS_INTERNAL_NULL_INPUT_ERR;
     }
 
-    if ( msKeyValStr->type == NULL ) {
-        fillStrInMsParam( msKeyValStr, NULL );
+    if (msKeyValStr->type == nullptr) {
+        fillStrInMsParam(msKeyValStr, nullptr);
     }
 
-    char *keyPtr = parseMspForStr( keyStr );
-    int keyLen;
-    if ( keyPtr == NULL || strcmp( keyPtr, MS_NULL_STR ) == 0 ) {
-        keyLen = 0;
-    }
-    else {
-        keyLen = strlen( keyPtr );
-    }
+    auto get_length{[](const char* str) -> size_t {
+        if (str == nullptr || std::strcmp(str, MS_NULL_STR) == 0) {
+	    return 0;
+	}
+	return std::strlen(str);
+    }};
 
-    char *valPtr = parseMspForStr( valStr );
-    int valLen;
-    if ( valPtr == NULL || strcmp( valPtr, MS_NULL_STR ) == 0 ) {
-        valLen = 0;
-    }
-    else {
-        valLen = strlen( valPtr );
-    }
-    if ( valLen + keyLen <= 0 ) {
+    const char *keyPtr{parseMspForStr(keyStr)};
+    int keyLen{static_cast<int>(get_length(keyPtr))};
+
+    const char *valPtr{parseMspForStr(valStr)};
+    int valLen{static_cast<int>(get_length(valPtr))};
+
+    if (valLen + keyLen <= 0) {
         return 0;
     }
 
-    char *oldKeyValPtr = parseMspForStr( msKeyValStr );
+    char *oldKeyValPtr{parseMspForStr(msKeyValStr)};
     char *newKeyValPtr, *tmpPtr;
-    if ( oldKeyValPtr == NULL ) {
-        int newLen = valLen + keyLen + 10;
-        newKeyValPtr = ( char * )malloc( newLen );
+    if (oldKeyValPtr == nullptr) {
+        const int newLen{valLen + keyLen + 10};
+        newKeyValPtr = reinterpret_cast<char*>(std::malloc(newLen));
```

Comment:
`reinterpret_cast` -> `static_cast`

For correctness: `newLen` -> `sizeof(char) * newLen`